libwnbd/libwnbd.cpp

```diff
@@ -235,6 +235,80 @@ DWORD WnbdRemoveEx(
     return Status;
 }
 
+DWORD WnbdSetDiskSize(PWNBD_DISK Disk, UINT64 BlockCount) {
+    DWORD ErrorCode = ERROR_SUCCESS;
+    DWORD BytesReturned = 0;
+
+    LogDebug("Resizing device %s. New block count: %d",
+             Disk->Properties.InstanceName, BlockCount);
+    if (!Disk->Handle || Disk->Handle == INVALID_HANDLE_VALUE) {
+        LogDebug("WNBD device not found");
+        return ERROR_NOT_FOUND;
+    }
+
+    ErrorCode = WnbdIoctlSetDiskSize(
+        Disk->Handle,
+        Disk->ConnectionInfo.ConnectionId,
+        BlockCount,
+        NULL);
+
+    if (ErrorCode) {
+        LogError("Could not set disk size.");
```

Comment:
we can skip this error message, we're already logging the error in ``WnbdIoctlSetDiskSize``.